### PR TITLE
Sync public chats

### PR DESCRIPTION
**Pull request by cammellos**
Add public chats to sync. Syncs only active public chats and does not syncs messages. Does not delete public chats.

`Sync all` will synchronize public chats open on device-A on device-B, similarly to the rest of the data.

If you join a public chat on device A it will also be joined on device B. 

### Testing

1) Open chat "#status" on device A
2) Pair devices A & B
3) Click on sync-all on device A
4) Chat #status should be open on device B
5) Open chat "#test-team" on device B
6) Chat "#test-team" should open on device A
7) Open chat "#status-core" on device A
8) Chat #stauts-core should open on device B

Known limitations:

1) Chat won't be synchronized if joined when offline, will be fixed in future pr

### Features added

It will also show now devices that haven't been paired but have been detected, icons will be updated to reflect that.

Prompt for syncing: Includes https://github.com/status-im/status-react/pull/6972 so they can be tested together
Limit number of devices to 3 (the one you are using + 2 extra)

status: ready

### CODE REVIEW

**Review by rasom — APPROVED** [review]

File: src/status_im/chat/models.cljs [hunk 1]
```diff
@@ -240,10 +240,11 @@
 
 (fx/defn start-public-chat
   "Starts a new public chat"
-  [cofx topic opts]
+  [cofx topic {:keys [dont-navigate?] :as opts}]
   (fx/merge cofx
             (add-public-chat topic)
-            (navigate-to-chat topic opts)
+            #(when-not dont-navigate?
```
Comment (yenda) [inline src/status_im/chat/models.cljs:246]: use `navigate?` to avoid double negation?

**Review by yenda — APPROVED** [review]

**Comment by annadanchenko** [comment]
@cammellos can you resolve conflicts in the branch, please? It's the reason bot is moving PR back from column TO TEST to CONTRIBUTOR

**Comment by annadanchenko** [comment]
@cammellos 
1. What's *active* public chat? Is it public chat that had at least 1 message in last 24 hours?
2. If user had syncing enabled on 0.9.32 then is it expected that public chats will be synced after upgrade to this PR?

**Comment by annadanchenko** [comment]
@cammellos upgraded 2 devices paired with each other: after upgrade paired devices show second device as enabled but no group chats are synchronised until I tap on Sync all devices. Not sure it's expected.

Steps:
1. pair device a and device b on 0.9.32 release
2. on device A join status chat, on device b join test-team chat
3. upgrade both devices
4. on device A check Devices list in Profile (in my case it shown Device B as enabled)
5. on device A Check if you have test-team chat shown
6. Only after tap on Sync all devices on Device A I can see chat "test-team" shown on device a and chat "status" on device B

**Comment by cammellos** [comment]
Thanks for taking the time to test it, @annadanchenko.
Correct, this is the expected behavior, for chats that have not been synced before you will have to `sync all`.

> What's active public chat? Is it public chat that had at least 1 message in last 24 hours?
If user had syncing enabled on 0.9.32 then is it expected that public chats will be synced after upgrade to this PR?

An active chat is just a public chat that is open and you can see it (sorry for the confusion, deleted chat are not active, but it's just an implementation detail for us developers).

As you pointed out they are synced only when `sync all` is pressed, but after that any new open chat will be shown on the other device.

Also if you now open a public chat on device A it should open on device B as well, if the two devices are paired and are running both this PR.

**Comment by annadanchenko** [comment]
@cammellos on iOS the toggle next to the Device name is shown as enabled while it wasn't.


Steps:
1. restore same account on Android and iOS devices
2. On android device enable Dev mode, open Devices, tap on Pair this device
3. As a result, on iOs device new item is shown in Paired devices with grey text but enabled toggle (this is an issue)
4. On android same item is shown as expected: if tap on Pair device on iOS then on Android device there is new item under Paired devices: grey text and disabled toggle
screenshot_20181220-102804

**Comment by annadanchenko** [comment]
I've done testing on build 6, Android, iOs and Windows. Public chats are synced between devices.
What's left:
1. check e2e results (on build 6 e2e build failed)
2. check on linux and macos
3. smoke check on already tested OS

**Comment by cammellos** [comment]
Issue with enabled devices should be resolved in the next build, I have swapped to checkboxes instead of switches on IOS.

**Comment by churik** [comment]
@cammellos 
The last thing - can you please highlight checkboxes somehow, because it is not obvious where is tappable area and user can get confused when trying to tap on disable icon in order to enable it?
1234t4654

And please rebase it to current develop branch in order to test linux image.

Thanks in advance!

**Comment by cammellos** [comment]
@churik rebased, if you don't mind I would work on checkboxes on a separate PR, as I am working on that area, is that ok?

**Comment by churik** [comment]
Retested on IOS and Mac OSx, tested on Linux Ubuntu and Windows 10 Home. 
Additionally tested sync group, 1-1 chats, contacts, photo and username.
Tested on chats with spam messages (>100k chars), amount of chats ~20.

**Pull request merged**